registration/include/pcl/registration/impl/ppf_registration.hpp

```diff
@@ -173,55 +182,59 @@ pcl::PPFRegistration<PointSource, PointTarget>::computeTransformation(
       }
     }
 
-    std::size_t max_votes_i = 0, max_votes_j = 0;
+    // the paper says: "For stability reasons, all peaks that received a certain amount
+    // of votes relative to the maximum peak are used." No specific value is mentioned,
+    // but 90% seems good
     unsigned int max_votes = 0;
-
-    for (std::size_t i = 0; i < accumulator_array.size(); ++i)
-      for (std::size_t j = 0; j < accumulator_array.back().size(); ++j) {
-        if (accumulator_array[i][j] > max_votes) {
+    const std::size_t size_i = accumulator_array.size(),
+                      size_j = accumulator_array.back().size();
+    for (std::size_t i = 0; i < size_i; ++i)
+      for (std::size_t j = 0; j < size_j; ++j) {
+        if (accumulator_array[i][j] > max_votes)
           max_votes = accumulator_array[i][j];
-          max_votes_i = i;
-          max_votes_j = j;
+      }
+    max_votes *= 0.9;
+    for (std::size_t i = 0; i < size_i; ++i)
+      for (std::size_t j = 0; j < size_j; ++j) {
+        if (accumulator_array[i][j] >= max_votes) {
+          Eigen::Vector3f model_reference_point = (*input_)[i].getVector3fMap(),
```

Comment:
Could have const in front of most of the variables here?

Reply:
@larshg I added const